tests/compute/test_sampler.py

```diff
@@ -155,10 +155,29 @@ def test_layer_sampler():
     _test_layer_sampler()
     _test_layer_sampler(prefetch=True)
 
+def test_setseed():
```

Comment:
@zheng-da @jermainewang @mufeili (since you're involved in a discussion thread)
I require that the NodeFlow sequence emitted by the sampler should be the same given the same seed, even with multiple workers.  I wonder if the requirement is appropriate or too strict, especially since we are specifying multiple workers.

Reply:
I think it's too strict. Basically, for multi-thread, use separate thread-local rng for each thread. It makes sure that each thread generates the same sequence but does not care about the concurrency.